rasa/core/channels/mattermost.py

```diff
@@ -16,12 +16,22 @@ class MattermostBot(MattermostAPI, OutputChannel):
     def name(cls):
         return "mattermost"
 
-    def __init__(self, url, team, user, pw, bot_channel):
+    @classmethod
+    def from_credentials(cls, credentials):
+        if not credentials:
+            cls.raise_missing_credentials_exception()
+
+        return cls(
+            credentials.get("webhook_url"),
+        )
+
+    def __init__(self, url, team, user, pw, bot_channel, webhook_url):
```

Comment:
Can you add types please? We add them step by step.  Thanks.

Reply:
Done (for new PR #4473 )